libcontainer/cgroups/fs2/hugetlb.go

```diff
@@ -39,15 +59,23 @@ func statHugeTlb(dirPath string, stats *cgroups.Stats) error {
 	}
 	hugetlbStats := cgroups.HugetlbStats{}
 
+	supportsReservationAccounting := HasReservationAccountingSupport(dirPath)
+
 	for _, pagesize := range hugePageSizes {
-		usage := strings.Join([]string{"hugetlb", pagesize, "current"}, ".")
+		filenamePrefix := strings.Join([]string{"hugetlb", pagesize}, ".")
+
+		if supportsReservationAccounting {
+			filenamePrefix += ".rsvd"
+		}
+
+		usage := fmt.Sprintf("%s.current", filenamePrefix)
 		value, err := fscommon.GetCgroupParamUint(dirPath, usage)
 		if err != nil {
 			return errors.Wrapf(err, "failed to parse hugetlb.%s.current file", pagesize)
 		}
 		hugetlbStats.Usage = value
 
-		fileName := strings.Join([]string{"hugetlb", pagesize, "events"}, ".")
+		fileName := fmt.Sprintf("%s.events", filenamePrefix)
```

Comment:
nit: using `fileName := filenamePrefix + ".events"` would be faster

but either way is fine